system/jlib/jfile.cpp

```diff
@@ -7579,3 +7589,103 @@ IAPICopyClient * createApiCopyClient(IStorageApiInfo * source, IStorageApiInfo *
     }
     return nullptr;
 }
+
+
+class CBlockedFileIO : public CSimpleInterfaceOf<IFileIO>
```

Comment:
Should have a comment that this is not thread safe, so can only be used from a single thread.

Reply:
will add comment